code/modules/vending/_vending.dm

```diff
@@ -1611,6 +1615,32 @@ GLOBAL_LIST_EMPTY(vending_machines_to_restock)
 	credits_contained = max(0, credits_contained - credits_to_remove)
 	SSblackbox.record_feedback("amount", "vending machine looted", holochip.credits)
 
+/obj/machinery/vending/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+	if(ishuman(user) && tilted && !held_item)
+		context[SCREENTIP_CONTEXT_LMB] = "Right the vending machine"
+		return TRUE
+
+	if(istype(held_item) && held_item?.tool_behaviour == TOOL_SCREWDRIVER)
+		context[SCREENTIP_CONTEXT_LMB] = panel_open ? "Close panel" : "Open panel"
+		return TRUE
+
+	if(panel_open && held_item && held_item?.tool_behaviour == TOOL_WRENCH)
+		context[SCREENTIP_CONTEXT_LMB] = anchored ? "Unsecure" : "Secure"
+		return TRUE
+
+	if(istype(held_item) && held_item.tool_behaviour == TOOL_CROWBAR && panel_open)
+		context[SCREENTIP_CONTEXT_LMB] = "Deconstruct"
+		return TRUE
+
+	if(istype(held_item) && (vending_machine_input[held_item.type] || canLoadItem(held_item, user, FALSE)))
+		context[SCREENTIP_CONTEXT_LMB] = "Load item into vending machine"
+		return TRUE
+
+	if(istype(held_item, refill_canister) && panel_open)
+		context[SCREENTIP_CONTEXT_LMB] = "Restock vending machine [credits_contained ? "and collect credits" : null ]"
+		return TRUE
```

Comment:
```suggestion
	if(panel_open && stype(held_item, refill_canister))
		context[SCREENTIP_CONTEXT_LMB] = "Restock vending machine [credits_contained ? "and collect credits" : null ]"
		return CONTEXTUAL_SCREENTIP_SET
```
Rearranging `panel_open` for consistency. 
Though all of the `panel_open` checks could probably be in the same conditional. 
Also: Tip might be a bit long? Maybe just leave it to "Restock supply" or something?

Reply:
I just noticed the `This also provides a context for restock modules, which shows specific text when the vending machine has credit to be claimed when restocked` part. 

Is this shown anywhere else? Like in `examine`? If not, you're hiding a bit of information from people who choose to disable tips.

Reply:
Didn't even know people *could* disable screentips. You're right to point that out, easy tweak.